dal/repositories/user_repository.py

```diff
@@ -0,0 +1,34 @@
+from controllers.auth import connect_to_db
+from models.user import User
+
+async def with_db_connection(func):
+    async def wrapper(*args, **kwargs):
+        await connect_to_db()
+        return await func(*args, **kwargs)
+    return wrapper
+
+@with_db_connection
```

Comment:
You only need to establish a connection to your MongoDB database once during your application's startup. This is typically done within an initialization function or a global configuration. Beanie handles managing the connection throughout your application, eliminating the need to repeatedly connect in every function.

Reply:
Beanie models (classes inheriting from Document) are independent of the database connection itself. This promotes cleaner code, easier testing (models can be tested without a database connection), and better maintainability.

Reply:
i uploaded again LMK